execution/gethexec/caff_node.go

```diff
@@ -81,12 +90,32 @@ func NewCaffNode(configFetcher SequencerConfigFetcher, execEngine *ExecutionEngi
 		return nil
 	}
 
+	db, err := stack.OpenDatabase("caffdata", 0, 0, "caffdata/", false)
+	if err != nil {
+		log.Crit("could not open the caff database")
+		return nil
+	}
+
+	currentPosition, nextHotshotBlock, err := ReadCurrentPositionAndNextHotshotBlockFromDb(db)
+	if err != nil {
+		log.Crit("failed to read current position and next hotshot block", "err", err)
+		return nil
+	}
+
+	if nextHotshotBlock == 0 {
+		// No next hotshot block found, so we need to start from config.CaffNodeConfig.NextHotshotBlock
+		nextHotshotBlock = config.CaffNodeConfig.NextHotshotBlock
+		currentPosition = 0
```

Comment:
Note: we should not set the currentPosition here, current position is already stored in the block database